src/marshmallow/fields.py

```diff
@@ -1901,30 +1873,58 @@ class EnumValue(Field):
         "unknown": "Must be one of: {choices}.",
     }
 
-    def __init__(self, cls_or_instance: Field | type, enum: type[Enum], **kwargs):
+    def __init__(
+        self,
+        enum: type[EnumType],
+        *,
+        by_value: bool | Field | type = False,
+        **kwargs,
+    ):
         super().__init__(**kwargs)
-        try:
-            self.field = resolve_field_instance(cls_or_instance)
-        except FieldInstanceResolutionError as error:
-            raise ValueError(
-                "The enum field must be a subclass or instance of "
-                "marshmallow.base.FieldABC."
-            ) from error
         self.enum = enum
-        self.choices = ", ".join(
-            [str(self.field._serialize(m.value, None, None)) for m in enum]
-        )
+        self.by_value = by_value
+
+        # Serialization by name
+        if by_value is False:
+            self.field: Field = String()
+            self.choices = ", ".join(
+                [str(self.field._serialize(m, None, None)) for m in enum.__members__]
```

Comment:
nit: intermediate list here is unnecessary

```suggestion
                str(self.field._serialize(m, None, None)) for m in enum.__members__
```

kinda surprised pyupgrade doesn't autofix this 🤔